src/main/java/com/iexec/core/task/update/TaskUpdateRequestManager.java

```diff
@@ -89,35 +109,62 @@ public void setRequestConsumer(final TaskUpdateRequestConsumer consumer) {
      * Retries consuming and notifying if interrupted
      */
     @Scheduled(fixedDelay = 1000)
+    @SneakyThrows
     public void consumeAndNotify() {
         if (consumer == null){
             log.warn("Waiting for consumer before consuming [queueSize:{}]", queue.size());
             return;
         }
 
-        while (!Thread.currentThread().isInterrupted()){
-            waitForTaskUpdateRequest();
+        // We should only start threads to complete the thread pool.
+        // This is useless for a fresh start but could be handy
+        // if current thread is stopped and current method is rescheduled.
+        final int nbThreadsToStart = TASK_UPDATE_THREADS_POOL_SIZE - updatingThreads.get();
```

Comment:
Step which takes a lot of time is `synchronized { consumer.onTaskUpdateRequest(..) }`, this why we were delegating it to a dedicated executor handling multiple updates at a time. Why now having multiple threads waiting-a-lot then racing-a-lot for next element of the queue?

Reply:
Thanks. I think your comment justifies the fact of putting  `queue.take()`  inside the async code block delegated to the pool executor, but why would we need 4 listening threads?

Reply:
That could maybe be something like: when starting a new update, check if we have some spare thread and start a new `queue.take()` if we do. At the end of each update, check if we still have at least one waiting thread. If we don't, start a new thread.
Does it look good to you?

Reply:
Don't we already have 1 waiting thread?

Reply:
I think there's a misunderstanding. I'm talking about the waiting threads blocked on `queue.take()`. They represent `TASK_UPDATE_THREADS_POOL_SIZE - updatingThreads` threads, waiting for a new task update request to be published.